scipy/optimize/tests/test_chandrupatla.py

```diff
@@ -493,7 +493,7 @@ def test_basic(self, p, xp):
         a, b = xp.asarray(-5.), xp.asarray(5.)
         res = _chandrupatla_root(self.f, a, b, args=(xp.asarray(p),))
         ref = xp.asarray(stats.norm().ppf(p), dtype=xp.asarray(p).dtype)
-        xp_assert_close(res.x, ref)
+        xp_assert_close(res.x, ref, check_0d=False)
```

Comment:
Is this an alternative?

```suggestion
        xp_assert_close(res.x, ref[()])
```

Reply:
I refrained from touching the reference value, because AFAIU it will then be wrong for non-numpy backends. It'd be possible to adapt the expectation in a `is_numpy(xp)` branch, but I think this is detrimental for readability, and also makes the inconsistency less visible.

Reply:
in the original PR this was discussed, and apparently NumPy is the only library that distinguishes between 0D arrays and scalars in this way, as per the comment on line 261 of this PR. It might be that a keyword is still preferable, but good to lay out all of the options.